src/EventStore.Core/Helpers/IODispatcher.cs

```diff
@@ -8,9 +8,12 @@
 using EventStore.Core.Services.AwakeReaderService;
 using EventStore.Core.Services.TimerService;
 using System.Collections.Generic;
+using System.Linq;
+using EventStore.Common.Utils;
 using EventStore.Core.Util;
 
 namespace EventStore.Core.Helpers {
+	
 	public sealed class IODispatcher : IHandle<IODispatcherDelayedMessage> {
```

Comment:
We could create a new `RequestTrackingIODispatcher` or have a switch that toggles tracking requests as it currently stands we are only interested in tracking requests for a single scenario and we likely don't want the overhead of tracking requests for everyone using the `IODispatcher`.

Reply:
Given that we are using the IODispatcher in a number of different places through the projections and it's not abstracted, I've changed it to use a switch to toggle tracking the requests.

However, I think it's worth revisiting this to pull out a separate dispatcher for projections at some stage